CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/base/BaseViewModel.kt

```diff
@@ -43,4 +47,8 @@ abstract class BaseViewModel : ViewModel() {
             _showLoading.value = false
         }
     }
+
+    fun execute(job: suspend () -> Unit) = viewModelScope.launch {
+        withContext(dispatchers.getIO()) { job.invoke() }
```

Comment:
I have a question please🙋‍♀️  ,Is it possible on some case we don't want to use `io` thread 🙏

Reply:
@Wadeewee In that case, you can create another `function`, like `executeOnMain` and get the corresponding `Dispatcher` from `dispatchers ` or provide the type of `Dispatcher` you want to do as the parameter of the `execute` function.
What do you think?

Reply:
@doannimble Providing a type via parameter is good to me 😃   but why we didn't inject the `dispatchers` in `UseCase` instead inject in `VM` so you can define the thread whatever you want in `UseCase` Please correct me If I missed something 🙇‍♀️

Reply:
@Wadeewee It such a good question. As I remember, it is related to the cancelation of suspend thread. But not exactly,
let me figure out the correct answer then explain to you soon 💯

Reply:
this seem to force every `job` to be executed always with `dispatchers.getIO()` 🤔  so in case we want to switch to default, the API doesn't seem to be very open for it; perhaps you can pass the `CoroutineDispatcher` in the params and default it as `IO()` instead?

Reply:
@sleepylee Okay, I think that is what @Wadeewee wants to. So let update to make it more open.